java/vector/src/main/java/org/apache/arrow/vector/dictionary/ValueMatcher.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.dictionary;
+
+import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.compare.RangeEqualsVisitor;
+
+/**
+ * A special {@link RangeEqualsVisitor} used to compare single values
+ * which used in dictionary encoding.
+ */
+public class ValueMatcher {
```

Comment:
I don't think this class is necessary

Reply:
@pravindra suggested hide the visitor API here to make it more clear, what do you think?

Reply:
We could also rename RangeEqualsVisitor impl to ValueMatcher.

Reply:
The only purpose of this class is make code readable, do you think it's ok to remove this as Bryan suggested?

Reply:
lgtm